lib/tests/test_revset.rs

```diff
@@ -222,78 +222,78 @@ fn test_resolve_symbol_change_id(readonly: bool) {
         "040031cb4ad0cbc3287914f1d205dabf4a7eb889"
     );
 
-    let repo;
-    let repo_ref = if readonly {
-        repo = tx.commit();
-        repo.as_repo_ref()
+    let mut _readonly_repo = None;
+    let repo: &dyn Repo = if readonly {
+        _readonly_repo = Some(tx.commit());
```

Comment:
`Option` isn't needed here. `_readonly_repo = tx.commit();` should be enough to keep it alive.

Reply:
Ah, good point. The variable is never used, so it doesn't need to be initialized in all code paths.